### PR TITLE
Rename all executables from cadence- to temporal-

### PR DESCRIPTION
**Pull request by samarabbas**
Rename generated binaries:
- cadence -> temporal
- cadence-server -> temporal-server
- cadence-canary -> temporal-canary
- cadence-sql-tool -> temporal-sql-tool
- cadecne-cassandra-tool -> temporal-cassandra-tool

Moved core schema files for all database implementation to be under temporal folder.

Also used 'temporal' as keyspace or database name for core schema.

Renamed all environment variable to use TEMPORAL instead of CADENCE.

Renamed all docker scripts to references to point to new executables and mention
temporal.

Change reference to buildkite pipelines for temporal.

### CODE REVIEW

File: cmd/server/temporal/temporal.go [hunk 1]
```diff
@@ -155,7 +155,7 @@ func constructPath(dir string, file string) string {
 func BuildCLI() *cli.App {
 
 	app := cli.NewApp()
-	app.Name = "cadence"
+	app.Name = "temporal"
 	app.Usage = "Cadence server"
```
Comment (alexshtin) [inline cmd/server/temporal/temporal.go]: `app.Usage`?

File: service/history/decisionChecker.go [hunk 2]
```diff
@@ -64,7 +64,7 @@ type (
 )
 
 const (
-	reservedTaskListPrefix = "/__cadence_sys/"
```
Comment (alexshtin) [inline service/history/decisionChecker.go]: Not part of this PR but this const seems to be defined in multiple places. Needs to be extracted somewhere.

File: cmd/server/temporal/temporal.go (hunk 1 above)
Comment (samarabbas) [reply cmd/server/temporal/temporal.go]: Let me do that in a separate change.

File: service/history/decisionChecker.go (hunk 2 above)
Comment (samarabbas) [reply service/history/decisionChecker.go]: Agreed.  Will do it in a separate PR.
